crates/muvm/src/monitor.rs

```diff
@@ -0,0 +1,94 @@
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::thread;
+use std::time;
+
+use anyhow::Result;
+use log::debug;
+use procfs::{Current, Meminfo};
+use uuid::Uuid;
+
+use crate::launch::request_launch;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum GuestPressure {
+    None,
+    Low,
+    Medium,
+    High,
+    Critical,
+}
+
+impl From<GuestPressure> for u32 {
+    fn from(pressure: GuestPressure) -> u32 {
+        match pressure {
+            GuestPressure::None => 10,
+            GuestPressure::Low => 1000,
+            GuestPressure::Medium => 2000,
+            GuestPressure::High => 3000,
+            GuestPressure::Critical => 3000,
+        }
+    }
+}
+
+pub fn spawn_monitor(server_port: u32, cookie: Uuid) {
+    thread::spawn(move || work(server_port, cookie));
+}
+
+fn set_guest_pressure(server_port: u32, cookie: Uuid, pressure: GuestPressure) -> Result<()> {
+    if pressure == GuestPressure::Critical {
+        debug!("requesting the guest to drop its caches");
+        // This is a fake command that tells muvm-server to write to "/proc/sys/vm/drop_caches"
+        let command = PathBuf::from("/muvmdropcaches");
+        let command_args = vec![];
+        let env = HashMap::new();
+        request_launch(server_port, cookie, command, command_args, env)?;
+    }
+
+    let wsf: u32 = pressure.into();
+    debug!("setting watermark_scale_factor to {wsf}");
+
+    let command = PathBuf::from("/sbin/sysctl");
+    let command_args = vec![format!("vm.watermark_scale_factor={}", wsf)];
+    let env = HashMap::new();
+    request_launch(server_port, cookie, command, command_args, env)
+}
+
+fn work(server_port: u32, cookie: Uuid) {
```

Comment:
Not a verb in this case (it'd have to be "do work")... What about `run`?

Reply:
`run` sounds good to me, changed.